src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -2667,6 +2667,56 @@ void java_lang_Throwable::get_stack_trace_elements(Handle throwable,
   }
 }
 
+Handle java_lang_Throwable::get_cause_with_stack_trace(Handle throwable, TRAPS) {
+  // Call to JVM to fill in the stack trace and clear declaringClassObject to
+  // not keep classes alive in the stack trace.
+  // call this:  public StackTraceElement[] getStackTrace()
+  assert(throwable.not_null(), "shouldn't be");
+
+  JavaValue result(T_ARRAY);
+  JavaCalls::call_virtual(&result, throwable,
+                          vmClasses::Throwable_klass(),
+                          vmSymbols::getStackTrace_name(),
+                          vmSymbols::getStackTrace_signature(),
+                          CHECK_NH);
+  Handle stack_trace(THREAD, result.get_oop());
+  assert(stack_trace->is_objArray(), "Should be an array");
+
+  // If the original exception was in the bootstrap class loader, then use that
```

Comment:
Not sure that is a sufficiently strong enough check due to -Xbootclasspath:/a. You may need to check actual java.base module.

Reply:
I don't know.  The worry was that the exception would be unloaded, which it won't if it's in the bootstrap class loader.

Reply:
There are multiple problems with instantiating an unknown instance of an exception type. The unloading aspect is only one of those problems. To be safe you really only want to instantiate a well-defined set of well-behaved exceptions types - and that would those (or a subset thereof) in the base module.

Maybe the safest thing is to always create an instance of EIIE to hold the original stacktrace and message. I actually find it semantically more appealing to have the NCDFE be caused by an EIIE.

Reply:
That might make things really simple and nice! I like that idea.